diff -ur a/lib/python3.4/site-packages/subdomains/compat/requestfactory.py b/lib/python3.4/site-packages/subdomains/compat/requestfactory.py
--- a/lib/python3.4/site-packages/subdomains/compat/requestfactory.py	2015-12-10 20:15:42.000000000 +0100
+++ b/lib/python3.4/site-packages/subdomains/compat/requestfactory.py	2015-10-07 15:56:15.938766725 +0200
@@ -3,7 +3,10 @@
 """
 import urllib
 from cStringIO import StringIO
-from urlparse import urlparse
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
 
 from django.conf import settings
 from django.core.handlers.wsgi import WSGIRequest
diff -ur a/lib/python3.4/site-packages/subdomains/utils.py b/lib/python3.4/site-packages/subdomains/utils.py
--- a/lib/python3.4/site-packages/subdomains/utils.py	2015-12-10 20:15:42.000000000 +0100
+++ b/lib/python3.4/site-packages/subdomains/utils.py	2015-10-08 12:12:05.340564253 +0200
@@ -1,5 +1,8 @@
 import functools
-from urlparse import urlunparse
+try:
+    from urlparse import urlunparse
+except ImportError:
+    from urllib.parse import urlunparse
 
 from django.conf import settings
 from django.contrib.sites.models import Site
